### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by mmatongo**
### Project requirements
- Create a class `Corrector` with a method `correct_name` that does the following:
  - Make sure the first letter of the given name is a capital letter.
  - Make sure that the name has a maximum of 10 characters. If it's longer it should trim the word.
- Set an instance of `Corrector` in `Person` on initialization.
- In `Person` add a method `validate_name` that uses the `Corrector` instance to validate and save the corrected name.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -16,4 +18,8 @@ def can_use_services?
   def of_age?
     @age >= 18
   end
+
+  def validate_name?
+    @name = Corrector.new.correct_name(@name)
```
Comment (SajjadAhmad14) [inline person.rb:23]: Please make an instance of the `Corrector` class in the Person class constructor and use that instance to call the `correct_name` method as it improves reusability.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Daniel, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git Flow is followed. :heavy_check_mark:
PR has a title and description. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Daniel, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

## Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Daniel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mmatongo** [comment]
> Hi Daniel,
> 
> Your project is complete! There is nothing else to say other than... it's time to merge it 🚀 : Congratulations! 🎉

Thanks

**Pull request merged**